esp/src/src/Graph7Widget.ts

```diff
@@ -0,0 +1,243 @@
+        this.inherited(arguments);
+        this.widget.MainBorderContainer.resize();
+    }
+
+    layout(args) {
+        this.inherited(arguments);
+    }
+
+    destroy(args) {
+        this.inherited(arguments);
+    }
+
+    //  Implementation  ---
+    _initGraphControls() {
+        aspect.after(registry.byId(this.id + "MainBorderContainer"), "resize", () => {
+            if (this._graph) {
+                this._graph
+                    .resize()
+                    .render()
+                    ;
+            }
+        });
+    }
+
+    _onRefresh() {
+        this.refreshData();
+    }
+
+    _onGraphRefresh() {
+        this._graph.data().subgraphs.forEach((sg: Subgraph) => {
+            sg.minState("normal");
+        });
+        delete this._graph["_prevLayout"];
+        this.loadGraph(w => {
+            this._graph.zoomToFit();
+        });
+    }
+
+    _onPartial(args) {
+        this._graph.data().subgraphs.forEach((sg: Subgraph) => {
+            sg.minState("partial");
+        });
+        this.loadGraph(w => {
+            this._graph.zoomToFit();
+        });
+    }
+
+    _onMax(args) {
+        this._graph.data().subgraphs.forEach((sg: Subgraph) => {
+            sg.minState("normal");
+        });
+        this.loadGraph(w => {
+            this._graph.zoomToFit();
+        });
+    }
+
+    _onZoomToFit(args) {
+        this._graph.zoomToFit();
+    }
+
+    _onZoomToWidth(args) {
+        this._graph.zoomToWidth();
+    }
+
+    _onZoomToPlus(args) {
+        this._graph.zoomPlus();
+    }
+
+    _onZoomToMinus(args) {
+        this._graph.zoomMinus();
+    }
+
+    isWorkunit() {
+        return lang.exists("params.Wuid", this);
+    }
+
+    isQuery() {
+        return lang.exists("params.QueryId", this);
+    }
+
+    init(params) {
+        if (this.inherited(arguments))
+            return;
+
+        this.initGraph();
+
+        this.doInit(params);
+    }
+
+    clear() {
+        this._graph
+            .data({vertices:[], edges:[]})
+            .render()
+            ;
+    }
+
+    doInit(wuid: string) {
+        this.wuid = wuid;
+
+        this.refreshData();
+    }
+
+    refreshData() {
+        if (this.isWorkunit()) {
+            return this.loadGraphFromWu(this.wuid, "", true);
+        } else if (this.isQuery()) {
+        }
+        return Promise.resolve();
+    }
+
+    loadGraphFromWu(wuid, graphName, refresh: boolean = false) {
+        return this.fetchScopeGraph(wuid, graphName, refresh).then(() => {
+            this.loadGraph();
+        });
+    }
+
+    initGraph() {
+        this.graphStatus = dom.byId(this.id + "GraphStatus");
+        this._graph = new GraphWidget()
+            .target(this.id + "MainGraphWidget")
+            .layout("Hierarchy")
+            .applyScaleOnLayout(true)
+            .showToolbar(false)
+            .allowDragging(false)
+            .on("progress", what => {
+                this.graphStatus.innerText = what === "end" ? "" : what;
```

Comment:
Translation needed

Reply:
Did you mean "Wait" here?

Reply:
This is just echoing messages from the Viz Framework - so translation may not be possible ATM.